site.yml

```diff
@@ -78,6 +78,36 @@
     - role: percona-backup
       tags: ['infra', 'percona']
 
+- name: install mongodb on primary
+  gather_facts: force
+  hosts: mongo_db[0]
+  vars:
+    mongodb_primary: True
+  roles:
+    - role: mongodb
+      tags: ['mongodb', 'mongodb-primary']
+      when: ceilometer.enabled|bool
+
+- name: install mongodb on secondaries
+  gather_facts: force
+  hosts: mongo_db:!mongo_db[0]
+  vars:
+    mongodb_primary: False
+  roles:
+    - role: mongodb
+      tags: ['mongodb', 'mongodb-secondary']
+      when: ceilometer.enabled|bool
```

Comment:
You already have a conditional set on the inclusion of `primary.yml`. That conditional is applied to every task within, so even if running in parallel the tasks would only execute on the primary node. I don't see a reason to break this into two role applications.